app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,129 @@
+class Course::RegistrationsController < Course::ModuleController
+  load_and_authorize_resource through: :course
+  before_action :ensure_course_open, :ensure_not_registered, :update_if_exist, only: :create
+
+  def index #:nodoc:
+    @staff_requests = @registrations.staff
+    @student_requests = @registrations.student
+  end
+
+  def create #:nodoc:
+    role = (CourseUser.roles.include?(params[:role])) ? params[:role] : :student
+    @registration.user = current_user
+    @registration.save!
+
+    # TODO: Notify lecturer
+    redirect_to course_path(@course), flash: { notice: t('.request_format') %
+      { role: t("course.registrations.role.#{role}") } }
+  end
+
+  def update #:nodoc:
+    if params[:update_single]
+      update_registration
+    elsif params[:approve_single]
+      approve_registration
+    elsif params[:delete_single]
+      redirect_to delete_course_registration_path(@course, @registration)
+    else
+      redirect_to course_path(@course), status: :bad_request
+    end
+  end
+
+  def delete #:nodoc:
+    message = t('.notice_format') % { role: t("course.registrations.role.#{@registration.role}"),
+                                      user: @registration.user.name }
+    @registration.destroy!
+    redirect_to course_registrations_path(@course), flash: { notice: message }
+  end
+
+  def update_multiple #:nodoc:
+    if should_process_single?
+      redirect_to update_course_registration_path(@course,
+                                                  params[:approve_single] || params[:delete_single],
+                                                  params)
+    else
+      process_multiple_registrations
+    end
+
+  end
+
+  private
+
+  def should_process_single? #:nodoc:
+    params[:approve_single] || params[:delete_single]
+  end
+
+  def should_process_all? #:nodoc:
+    params[:approve_all] || params[:delete_all]
+  end
+
+  def should_approve? #:nodoc:
+    params[:approve_single] || params[:approve_selected] || params[:approve_all]
+  end
+
+  def update_scope #:nodoc:
+    params[:approve_all] || params[:delete_all]
+  end
+
+  def update_registration #:nodoc:
+    @registration.update_attributes(registration_params)
+
+    redirect_to course_path(@course),
+                flash: { notice: t('.update_notice_format') %
+                  { role: t("course.registrations.role.#{@registration.role}") } }
+  end
+
+  def approve_registration #:nodoc:
+    flash = {}
+    begin
+      flash[:notice] = t('.approve_notice_format') %
+        { role: t("course.registrations.role.#{@registration.role}"),
+          user: @registration.user.name }
+      Course::Registration.process!(@course, [@registration.id], approve: true)
+    rescue
+      flash[:error] = t('.error_format') % { reason: $ERROR_INFO }
+    end
+
+    redirect_to course_registrations_path(@course), flash: flash
+  end
+
+  def process_multiple_registrations
```

Comment:
Assignment Branch Condition size for process_multiple_registrations is too high. [18.36/15]